### PR TITLE
Added more great Symfony 4.1 features

### PR DESCRIPTION
**Pull request by javiereguiluz**
The last remaining big features are:

* Messenger component: is being discussed in #808.
* Advanced Console Output: I plan to send another PR with some changes in some commands.

### CODE REVIEW

File: src/Controller/BlogController.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ class BlogController extends AbstractController
     /**
      * @Route("/", defaults={"page": "1", "_format"="html"}, methods={"GET"}, name="blog_index")
      * @Route("/rss.xml", defaults={"page": "1", "_format"="xml"}, methods={"GET"}, name="blog_rss")
-     * @Route("/page/{page}", defaults={"_format"="html"}, requirements={"page": "[1-9]\d*"}, methods={"GET"}, name="blog_index_paginated")
+     * @Route("/page/{page<[1-9]\d*>}", defaults={"_format"="html"}, methods={"GET"}, name="blog_index_paginated")
```
Comment (stof) [inline src/Controller/BlogController.php]: I would even change it to `\d*+`. Using possessive quantifiers in routing requirements makes matching faster

File: src/Controller/Admin/BlogController.php [hunk 2]
```diff
@@ -130,7 +130,7 @@ public function show(Post $post): Response
     /**
      * Displays a form to edit an existing Post entity.
      *
-     * @Route("/{id}/edit", requirements={"id": "\d+"}, methods={"GET", "POST"}, name="admin_post_edit")
+     * @Route("/{id<\d+>}/edit",methods={"GET", "POST"}, name="admin_post_edit")
```
Comment (stof) [inline src/Controller/Admin/BlogController.php]: I suggest using a possessive quantifier here (`\d++`), to improve the route matching performance (Symfony generates possessive quantifiers for its builtin requirements, but custom ones need to do that themselves, as using non-possessive quantifiers can still be a valid use case)

File: src/Entity/User.php [hunk 3]
```diff
@@ -42,20 +43,24 @@ class User implements UserInterface, \Serializable
...
+     * @Assert\NotBlank()
      */
     private $fullName;
 
     /**
      * @var string
      *
      * @ORM\Column(type="string", unique=true)
+     * @Assert\NotBlank()
+     * @Assert\Length(min = 2, max = 50)
```
Comment (stof) [inline src/Entity/User.php]: The existing annotations are not putting spaces around the `=`

**Comment by javiereguiluz** [comment]
@stof I've merged without making the changes in the routing requirements ... but I've opened #818 to discuss about your proposal. Cheers!

**Pull request closed without merging**